lib/livebook/ecto_types/secret_origin.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Livebook.EctoTypes.SecretOrigin do
+  @moduledoc false
+  use Ecto.Type
+
+  @type t :: :system_env | :startup | :app | <<_::64, _::_*8>>
```

Comment:
Can we namespace the binary? Something like `{:enterprise, id}`.

Reply:
I think we can only store strings. So it would need to be "enterprise-#{id}" and then I am not sure if it is worth it.

Reply:
Prefixing sounds good to me actually :)

Reply:
That's for storage only right? Then we load it as `{:enterprise/hub, id}`.

Reply:
Also, what do you think about calling it `:session` instead of `:system_env`? That's a better representation, no? `:system_env` can slightly get confused with `:startup` (since startup is set via system_env).

Reply:
Good call!

Reply:
> That's for storage only right? Then we load it as {:enterprise/hub, id}.

Ah, good call!